### PR TITLE
:zap: discover assets in parallel

### PR DESCRIPTION
**Pull request by afiune**
Depends on https://github.com/mondoohq/cnquery/pull/4970

This PR is using the new `workerpool` package to discover assets in parallel.

There were a few places that I detected race conditions that I have fixed, I
also added a race detector `make` command which is running as a CI job.

## Testing

Using the `github` provider for testing, scanning an **organization that has around 3k repositories**.

### Before (~15 Minutes)
```
TRC logger.FuncDur> func=explorer.discoverAssets took=890303.455042
```
### After (~2 minutes)
```
TRC logger.FuncDur> func=explorer.discoverAssets took=124293.542166
```

## Race Detector

You can now run `make race/go` to check for race conditions.
```
$ make race/go
go test -race go.mondoo.com/cnquery/v11/internal/workerpool
ok  	go.mondoo.com/cnquery/v11/internal/workerpool	16.417s
go test -race go.mondoo.com/cnquery/v11/explorer/scan
ok  	go.mondoo.com/cnquery/v11/explorer/scan	2.487s
```

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -198,6 +198,18 @@
         "shell", "ssh", "user@18.215.249.49",
       ],
     },
+    {
+      "name": "scan github org",
```
Comment (jaym) [inline .vscode/launch.json:202]: did you mean to keep this in here

Comment (afiune) [reply .vscode/launch.json:202]: Yes. I think it is useful for folks testing, specially now that we have https://github.com/hit-training.

I can remove it.

**Review by jaym — APPROVED** [review]

**Comment by afiune** [comment]
You can now run `make race/go`
```
$ make race/go
go test -race go.mondoo.com/cnquery/v11/internal/workerpool
ok  	go.mondoo.com/cnquery/v11/internal/workerpool	16.417s
go test -race go.mondoo.com/cnquery/v11/explorer/scan
ok  	go.mondoo.com/cnquery/v11/explorer/scan	2.487s
```

I'll add a CI job for it.

File: internal/workerpool/pool.go [hunk 2]
```diff
@@ -92,7 +116,7 @@ func (p *Pool[R]) Close() {
 
 // Wait waits until all tasks have been processed.
 func (p *Pool[R]) Wait() {
-	ticker := time.NewTicker(100 * time.Millisecond)
+	ticker := time.NewTicker(10 * time.Millisecond)
```
Comment (afiune) [inline internal/workerpool/pool.go:119]: This fixed the benchmark failure.

File: providers/github/connection/connection.go [hunk 3]
```diff
@@ -74,6 +74,7 @@ func NewGithubConnection(id uint32, asset *inventory.Asset) (*GithubConnection,
 	ctx := context.WithValue(context.Background(), github.SleepUntilPrimaryRateLimitResetWhenRateLimited, true)
 
 	// perform a quick call to verify the token's validity.
+	// @afiune do we need to validate the token for every connection? can this be a "once" operation?
```
Comment (afiune) [inline providers/github/connection/connection.go:77]: This is fixed with https://github.com/mondoohq/cnquery/pull/4980

File: providers/runtime.go [hunk 4]
```diff
@@ -232,7 +235,9 @@ func (r *Runtime) Connect(req *plugin.ConnectReq) error {
 
 	// }
 
+	r.mu.Lock()
 	r.Provider.Connection, r.Provider.ConnectionError = r.Provider.Instance.Plugin.Connect(req, &callbacks)
+	r.mu.Unlock()
```
Comment (afiune) [inline providers/runtime.go]: The problem is the assignation here, not the actual gRPC call `Connect()`. I will change this.

Comment (afiune) [reply providers/runtime.go]: ✅  updated

**Review by imilchev — APPROVED** [review]

**Pull request merged**